openlibrary/plugins/worksearch/schemes/works.py

```diff
@@ -344,7 +347,11 @@ def convert_work_field_to_edition_field(
                     return WORK_FIELD_TO_ED_FIELD[field]
                 elif field.startswith('id_'):
                     return field
-                elif field in self.all_fields or field in self.facet_fields:
+                elif (
+                    self.is_search_field(field)
+                    or field in self.all_fields
+                    or field in self.facet_fields
+                ):
```

Comment:
`all_fields` is no longer needed! THat's done by `is_search_field`
```suggestion
                elif self.is_search_field(field) or field in self.facet_fields:
```